### PR TITLE
Feature/package dependencies

### PR DESCRIPTION
**Pull request by alan-cole**
**JIRA issue:** No ticket.

### Changed
I noticed there was an inconsistency with component package.json dependencies. Many packages that were dependencies were not correctly included to package.json files.

This PR adds automatic dependency inclusion for ripple packages, by sniffing out ripple import statements in .vue files. It leverages on existing code from `generate-dependencies.js`.

This automated script can be triggered by running `npm run package-dependencies`. Running `npm run docs` after that will update all readme files.

Doing this uncovered a lot of imports (like RplIcon) that were no longer used, but never removed, so these unneeded dependencies were also updated on this ticket.

The 4 commits were separated to make reading the changes easier (as a lot of files changed).

1. Add script to check ripple component dependencies.
-> This is the new script.
2. Update package dependencies / docs.
-> No new code here, but this does show the missing dependencies.
3. Removing unused imports.
-> This is the cleanup up existing components.
4. Updating dependencies and docs. 
-> No new code here, just an update after component cleanup.

### CODE REVIEW

**Comment by tim-yao** [comment]
@alan-cole Looks nice to have it. But I will back to CR this one after we released 15, feel there is a lot changes here, not have too much time atm. cheers!

**Comment by alan-cole** [comment]
@tim-yao I've updated this one to keep it relevant.

File: packages/Organisms/SiteHeader/README.md [hunk 1]
```diff
@@ -20,7 +20,11 @@ npm install @dpc-sdp/ripple-site-header --save
 ├── @dpc-sdp/ripple-icon
 ├── @dpc-sdp/ripple-link
 ├── @dpc-sdp/ripple-meta-tag
+<<<<<<< HEAD
 └── body-scroll-lock
+=======
+└── @dpc-sdp/ripple-search
+>>>>>>> Update package dependencies / docs.
```
Comment (tim-yao) [inline packages/Organisms/SiteHeader/README.md]: Ooops 🤓

File: scripts/generate-dependencies/generate-dependencies.js [hunk 2]
```diff
@@ -1,46 +1,27 @@
...
- *   1. A `package.json` file with:
- *     - name
+ * 1. A `package.json` file with:
+ *   - name
+ * 2. A `lerna.json` file with:
+ *   - version
  *
  * Script expects to be called via npm in the root directory:
  *
  *   `npm run package-dependencies`
```
Comment (tim-yao) [inline scripts/generate-dependencies/generate-dependencies.js]: Hi @alan-cole could you please add this into CONTRIBUTE.md.

File: packages/Organisms/SiteHeader/README.md (hunk 1 above)
Comment (alan-cole) [reply packages/Organisms/SiteHeader/README.md]: Woops. Fixed.

File: scripts/generate-dependencies/generate-dependencies.js (hunk 2 above)
Comment (alan-cole) [reply scripts/generate-dependencies/generate-dependencies.js]: Added requirements to contribute.md

**Comment by alan-cole** [comment]
Resolved issues. Updated new packages / documentation.

**Pull request merged**